### PR TITLE
bumped to zio 0.18, modified code accordingly

### PR DESCRIPTION
**Pull request by sirocchj**
main change is `IO[Exception, A] => Task[A]` // i.e. `IO[Throwable, A] == ZIO[Any, Throwable. A]`

### CODE REVIEW

File: zio/src/main/scala/log/effect/zio/ZioLogWriter.scala [hunk 1]
```diff
@@ -78,7 +76,7 @@ object ZioLogWriter {
 
   implicit final private def instance[E]: EffectSuspension[IO[E, ?]] =
     new EffectSuspension[IO[E, ?]] {
-      def suspend[A](a: =>A): IO[E, A] = IO.sync(a)
+      def suspend[A](a: =>A): IO[E, A] = IO.suspend(IO.succeed(a))
```
Comment (sirocchj) [inline zio/src/main/scala/log/effect/zio/ZioLogWriter.scala]: @barambani  ^^

Comment (barambani) [reply zio/src/main/scala/log/effect/zio/ZioLogWriter.scala]: to respect the semantic that `suspend` accepts any thunk (side effecting ones) we could go with something like
```scala
implicit final private val instance: EffectSuspension[Task] =
  new EffectSuspension[Task] {
    def suspend[A](a: =>A): Task[A] = IO.effect(a)
  }
```
this means that we have to change also `consoleLogZio` to return a `Task` and [this](https://github.com/laserdisc-io/log-effect/blob/master/zio/src/test/scala/ConsoleLogWriterTest.scala#L11) to 
```scala
private def capturedConsoleOutOf(write: Task[Unit]): String
```

File: zio/src/test/scala/ConsoleLogWriterTest.scala [hunk 2]
```diff
@@ -262,5 +262,5 @@ final class ConsoleLogWriterTest extends WordSpecLike with Matchers with zio.App
     }
   }
 
-  def run(args: List[String]): IO[Nothing, ExitStatus] = IO.succeed(ExitStatus.ExitNow(0))
+  def run(args: List[String]): IO[Nothing, Int] = IO.done(Exit.succeed(0))
```
Comment (barambani) [inline zio/src/test/scala/ConsoleLogWriterTest.scala]: the return type of `App.run` now is
```scala
def run(args: List[String]): ZIO[Environment, Nothing, Int]
```

Comment (sirocchj) [reply zio/src/test/scala/ConsoleLogWriterTest.scala]: added `override` too 😛

**Comment by barambani** [comment]
go for it

**Pull request merged**